sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/models/NtfsFileAttributes.java

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.models;
+
+import com.azure.storage.common.Constants;
+
+import java.util.EnumSet;
+
+/**
+ * Specifies options for NTFS File Attributes.
+ */
+public enum NtfsFileAttributes {
+
+    /**
+     * The file is read-only.
+     */
+    READ_ONLY,
+
+    /**
+     * The file is hidden, and thus is not included in ordinary directory listing.
+     */
+    HIDDEN,
+
+    /**
+     * The file is a system file. That is, the file is part of the operating system
+     * or is used exclusively by the operating system.
+     */
+    SYSTEM,
+
+    /**
+     * The file is a standard file that has no special attributes.
+     */
+    NORMAL,
+
+    /**
+     * The file is a directory.
+     */
+    DIRECTORY,
+
+    /**
+     * The file is a candidate for backup or removal.
+     */
+    ARCHIVE,
+
+    /**
+     * The file is temporary. A temporary file contains data that is needed while an
+     * application is executing but is not needed after the application is finished.
+     * File systems try to keep all the data in memory for quicker access rather than
+     * flushing the data back to mass storage. A temporary file should be deleted by
+     * the application as soon as it is no longer needed.
+     */
+    TEMPORARY,
+
+    /**
+     * The file is offline. The data of the file is not immediately available.
+     */
+    OFFLINE,
+
+    /**
+     * The file will not be indexed by the operating system's content indexing service.
+     */
+    NOT_CONTENT_INDEXED,
+
+    /**
+     * The file or directory is excluded from the data integrity scan. When this value
+     * is applied to a directory, by default, all new files and subdirectories within
+     * that directory are excluded from data integrity.
+     */
+    NO_SCRUB_DATA;
+
+    /**
+     * Converts an enum set of {@code NtfsFileAttributes} to a string.
+     *
+     * @return A <code>String</code> that represents the NTFS Attributes in the correct format delimited by |
+     *         which is described at {@link #toAttributes(String)}.
+     */
+    public static String toString(EnumSet<NtfsFileAttributes> ntfsAttributes) {
+        if (ntfsAttributes == null) {
+            return Constants.EMPTY_STRING;
+        }
+
+        final StringBuilder builder = new StringBuilder();
+
+        if (ntfsAttributes.contains(NtfsFileAttributes.READ_ONLY)) {
```

Comment:
Could add a small helper function which takes in the StringBuilder, EnumSet, Enum to look for, and String to add if enum is found.